dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -271,6 +271,19 @@ def replace_hostname(url)
   @browser.execute_script %q-sessionStorage.setItem('progress', '{"hourofcode":{"' + appOptions.serverLevelId + '":100}}')-
 end
 
+Then /^validate progress for level is "([^"]*)"/ do |test_result|
+  steps %{
+    And I reload the page
+    And I wait to see ".header_middle"
+    And I wait for 10 seconds
+    And element ".header_middle a.level_link:last" has class "#{test_result}"
+    Then I am on "http://studio.code.org/s/hourofcode"
```

Comment:
Suggestion: Break the second part of this (validating level progress on the script overview page) into its own step.  Then you can reuse the "validate progress for level" part in more places in the scenario, and be explicit about the "validate progress on script overview" as the last step in the scenario.